### ISSUE TITLE
Image orientation

### ISSUE THREAD

**Issue by hutou** [seebye/ueberzug#99]
It would be great if Exif data (Orientation) could be taken into account.
Easy or not ?

**Comment by seebye** [comment]
tl;dr The information isn't available.  
Also I'm not sure whether ueberzug should transform images on it's own.  
If you're using ranger there's an option to do that in the preview script.  
https://github.com/ranger/ranger/blob/master/ranger/data/scope.sh#L142-L150

I encountered a bug on implementing parallel image decoding.  
Python isn't able to execute code in parallel by using threads due to the global interpreter lock
(as a matter of fact c extension code can be executed in parallel).  
That's why the usage of multiple processes is required.  
As multiple processes are used the decoded images needs to be transfered between these processes.  
Unfortunately I encountered a bug of pillow during testing the implementation which caused some PNGs to be reported as webp.  
This bug leads to crashs.  
So I implemented a workaround by reducing the transfered information (the bug seemed to be unrelated to the data ueberzug requires):  
https://github.com/seebye/ueberzug/commit/db97f243788a287abda8b0fde3d13a1f1015b620
which is based on pillows implementation:
https://github.com/python-pillow/Pillow/blob/bd0f7b7116c956a0ce2afdf8970f69b656d35a11/src/PIL/Image.py#L691

I don't expect this bug to be fixed as I didn't report the issue and also I don't know anymore which image it was I encoutered the issue with.  
Unfortunately the removed information contains the exif data, so these informations aren't available to the main process of ueberzug.  
However I'm not sure but I think the problem occured on restoring the palette data not the meta data.  
Also I'm not sure whether ueberzug should transform images on it's own.

**Comment by cubetastic33** [comment]
> but I'm not sure whether ueberzug should transform images on its own
Doesn't it already transform the image with the ImageScalers?

Also - one way I can think of for applying transforms that ueberzug does not yet support is to create a transformed image using other tools, like imagemagick or pillow, write it to a temporary path (like `/tmp/pic.png`), and give that path to ueberzug. This method can probably be used to rotate the image if you're not using ranger.

**Comment by seebye** [comment]
>  Doesn't it already transform the image with the ImageScalers?  

Ueberzug doesn't do this **on it's own** it's asked to do so.  
Also it's necessary to define a way to fill the space the image should be place in.
In opposite to this ueberzug was never asked to rotate an image so this is something else.  
I'm not sure whether as developer I would expect it to apply these transformations and if I wouldn't I would likely be pissed about it.

**Comment by cubetastic33** [comment]
Ah - that makes complete sense, I think I'd just misunderstood what you meant. So maybe there could be a parameter that specifies if exif data should be considered?

**Comment by grilledcarpet** [comment]
I too am wondering about this. It'd be the icing on the cake really...
